### PR TITLE
Color extension

### PR DESCRIPTION
**Pull request by 3Qax**
Resolved issue #1

### CODE REVIEW

File: NetguruCollegeApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		19BB056323115DF000E4E342 /* UIColor.swift in Sources */ = {isa = PBXBuildFile; fileRef = 19BB056223115DF000E4E342 /* UIColor.swift */; };
```
Comment (anyashka) [inline NetguruCollegeApp.xcodeproj/project.pbxproj]: I would name it `UIColor+Extension.swift` as this is actually not a `UIColor` class in there, so it helps for a better understanding.

File: Project/Extensions/UIColor.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
+//
+//  UIColor.swift
+//  NetguruCollegeApp
+//
+import UIKit
+
+extension UIColor {
```
Comment (anyashka) [inline Project/Extensions/UIColor.swift]:
```suggestion
extension UIColor {

```

File: Project/Extensions/UIColor.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        case 8:
+            alpha = CGFloat(strtoul(String(chars[0...1]), nil, 16)) / 255
+            red = CGFloat(strtoul(String(chars[2...3]), nil, 16)) / 255
+            green = CGFloat(strtoul(String(chars[4...5]), nil, 16)) / 255
+            blue = CGFloat(strtoul(String(chars[6...7]), nil, 16)) / 255
+        default:
+            return nil
+        }
+        self.init(red: red, green: green, blue: blue, alpha: alpha)
+    }
```
Comment (anyashka) [inline Project/Extensions/UIColor.swift]:
```suggestion
    }
    
```

File: Project/Extensions/UIColor.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    static let defaultGreen = UIColor(hexString: "2AC503")
```
Comment (anyashka) [inline Project/Extensions/UIColor.swift]: Usually, we define the variables above all methods.

Also, if the project would be a bigger one is also a nice practice to implement colors in another class which holds colors used inside an application.

File: Project/Extensions/UIColor.swift [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    ///
+    /// - Parameters:
+    ///   - hexString: a hex representation of color. 3, 6 or 8 characters with or without a '#'
+    convenience init?(hexString: String) {
+        var chars = Array(hexString.hasPrefix("#") ? hexString.dropFirst() : hexString[...])
+        let red, green, blue, alpha: CGFloat
+        switch chars.count {
+        case 3:
+            chars = chars.flatMap { [$0, $0] }
+            fallthrough
```
Comment (anyashka) [inline Project/Extensions/UIColor.swift]: Why do you need `fallthrough`? :)

File: Project/Extensions/UIColor.swift [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+//  UIColor.swift
+//  NetguruCollegeApp
+//
+import UIKit
+
+extension UIColor {
+    /// Creates an instance of UIColor from given hexadecimal number given as a String.
+    /// Fails if given parameter is inncorect hexadecimal number.
+    ///
+    /// - Parameters:
```
Comment (anyashka) [inline Project/Extensions/UIColor.swift]: The documentation with one parameter can be written like:
``` /// - Parameters variable:  This does bla bla ```

File: Project/Extensions/UIColor.swift [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    ///   - hexString: a hex representation of color. 3, 6 or 8 characters with or without a '#'
+    convenience init?(hexString: String) {
```
Comment (anyashka) [inline Project/Extensions/UIColor.swift]: Can you please explain this part? How did exactly you implement?
What do you think about this implementation:

```    /// Initializes and returns a color object using the hex value.
    /// Parameter hex: The hex value of the color object.
    /// Returns: The color object.
    static func with(hex: Int) -> UIColor {
        return color(red:(hex >> 16) & 0xff, green:(hex >> 8) & 0xff, blue:hex & 0xff)
    }


    /// Initializes and returns a color object using the specified opacity and RGB component values.
    /// - Parameters:
    ///     - red: The red value of the color object.
    ///     - green: The green value of the color object.
    ///     - blue: The blue value of the color object.
    ///     - alpha: The opacity value of the color object.
    /// Returns: The color object.
    static func color(red: Int, green: Int, blue: Int, alpha: Int = 0xFF) -> UIColor {
        assert(red >= 0 && red <= 255, "Invalid red component")
        assert(green >= 0 && green <= 255, "Invalid green component")
        assert(blue >= 0 && blue <= 255, "Invalid blue component")
        return UIColor(red: CGFloat(red) / 255.0, green: CGFloat(green) / 255.0, blue: CGFloat(blue) / 255.0, alpha: CGFloat(alpha) / 255.0)
    }
```

**Review by anyashka — CHANGES_REQUESTED** [review]

File: Project/Extensions/UIColor+Extension.swift [hunk 8]
```diff
@@ -0,0 +1,33 @@
+//
+//  UIColor.swift
+//  NetguruCollegeApp
+//
+import UIKit
+
+extension UIColor {
+    
+    static let defaultGreen = UIColor.with(hex: 0x2AC503)
+    static let defaultDarkerGreen = UIColor.with(hex: 0x21A700)
```
Comment (anyashka) [inline Project/Extensions/UIColor+Extension.swift]: What about just `defaultDarkGreen`? :)

**Review by anyashka — COMMENTED** [review]
🙌🏻great

File: Project/Extensions/UIColor+Extension.swift (hunk 8 above)
Comment (3Qax) [reply Project/Extensions/UIColor+Extension.swift]: Good catch 👏

**Review by anyashka — APPROVED** [review]

**Pull request merged**